# Notebook Instructions
<i>You can run the notebook document sequentially (one cell a time) by pressing <b> shift + enter</b>. While a cell is running, a [*] will display on the left. When it has been run, a number will display indicating the order in which it was run in the notebook [8].</i>

<i>Enter edit mode by pressing <b>`Enter`</b> or using the mouse to click on a cell's editor area. Edit mode is indicated by a green cell border and a prompt showing in the editor area.</i>

# Call price using Vega

In this notebook, we will calculate approximate price of call option using Vega. The formula to compute the call option price is given as:

<i><b>Price of a call option = Initial Option Price + Vega x (change in the Volatility) * 100</b></i>



# Vega when implied volatility is 30%

In [5]:
# Import Library
import mibian
from tabulate import tabulate

# Define parameters for call option contract
stock_price = 100
strike_price = 100
interest_rate = 0
days_to_expiry = 40
volatility = 30

# Compute greeks
greeks_stock_price_100 = mibian.BS([stock_price, strike_price, interest_rate, days_to_expiry], \
                   volatility=volatility)

# Print Vega
print tabulate(
        [             
            ['Stock Price', 100],
            ['Call Price', greeks_stock_price_100.callPrice],
            ['Vega', greeks_stock_price_100.vega]                       
        ], 
        tablefmt='orgtbl')

| Stock Price | 100        |
| Call Price  |   3.96038  |
| Vega        |   0.131904 |


# Call price using Vega when implied volatility is 31%

In [7]:
call_price_at_31 = greeks_stock_price_100.callPrice \
                  + greeks_stock_price_100.vega * (0.31 - 0.30) * 100
                  
print call_price_at_31

4.09228021743


# Actual call price when implied volatility is 31%

In [8]:
volatility = 31
actual_call_price_at_31 = mibian.BS([stock_price, strike_price, interest_rate, days_to_expiry], \
                   volatility=volatility).callPrice

print actual_call_price_at_31

4.09227473664


Thus the call price computed using Vega of 4.09228 is pretty much close to the actual price of 4.09227. The call price is not equal to actual price due to the fact that there are higher order Greeks than Vega. But as seen here they do not have much of an impact.